### DISCUSSION TITLE
[2.5D Map] Building zoom level

### DISCUSSION THREAD

**Discussion by wolfgang-ch** [mytourbook/mytourbook#859] [General]
### Issue https://github.com/mytourbook/mytourbook/issues/867

The default zoom min level for building is 18. When reducing (hacking) this level to 16 then the buildings are still OK and OpenGL is still fast enough but when the zoom level is set to 15 then this issue occure, only the roof is visible on some buildings but not the building walls

New York with zoomlevel 16
![image](https://user-images.githubusercontent.com/1283445/167561026-42802a4f-045a-4a69-86b4-9e7f267db547.png)

**Adjusted code**
These code adjustments are very critical, when only a part is modified, it can happen that no building is displayed !

_net.tourbook_

![image](https://user-images.githubusercontent.com/1283445/167562685-6d7c19ea-d066-49e9-8e95-22f0bea60d96.png)

![image](https://user-images.githubusercontent.com/1283445/167562723-15ca12ec-f445-4ed0-94c3-a3e11580642e.png)

_vtm-parent_

![image](https://user-images.githubusercontent.com/1283445/167562758-a92f912a-17cb-4065-aa65-6d8d0f29dd65.png)

vtm-parent plugin: https://github.com/mytourbook/mytourbook/issues/516#issuecomment-1112904183

**Comment by telemaxx** [comment]
i had never understand why i have to zoom in that much to see the buildings.
so it looks real cool.

i am not sure if that low zoom level also works with offline maps.
a while ago asked emux about zoomlevel and buildings.
when i remember correctly he said the maps must also adjusted to see buildings i lower zoom levels.
but not sure.

**Comment by wolfgang-ch** [comment]
I played more with this min zoom level and below 15 there can be issues that the map is drawn wrongly.

The 2.5D map developers seems not to be willing to fix it https://github.com/mapsforge/vtm/discussions/927#discussioncomment-2727896 after I tested it as requested.

**Reply by telemaxx** [reply]
but with 15/16 it is also a big improvement.

**Reply by wolfgang-ch** [reply]
15/16 is now available in the main branch.

I had to use Java reflection because the 2.5 map author do not want to change the code https://github.com/mapsforge/vtm/discussions/927#discussioncomment-2735903
